### PR TITLE
[TS] LPS-150759

### PR DESCRIPTION
**Pull request by rolandpakai**
Hi Team,

could you please review my fix?

**Steps to reproduce:**

1. Start the bundle 
2. Create a Vocabulary, make it required for Pages
3. Create a Category in it
4. Create a Content page, select the category
5. Go to its Look and feel configuration and try to save it

**Current behavior:** You won't be able to do that, "Error:Your request failed to complete."

**Expected behavior**: You should be able to save the configuration

Solution
Based on the LayoutDetailsFormNavigatorEntry::isVisible logic we set the assetCategoryIds in the context in case the details form navigation entry is not visible.

https://github.com/liferay/liferay-portal/blob/master/modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/frontend/taglib/form/navigator/LayoutDetailsFormNavigatorEntry.java#L49-L57

Thanks, Roland

https://issues.liferay.com/browse/LPS-150759

### CODE REVIEW

File: modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/portlet/action/EditLayoutMVCActionCommand.java [hunk 1]
```diff
@@ -124,6 +126,16 @@ protected void doProcessAction(
 			ServiceContext serviceContext = ServiceContextFactory.getInstance(
 				Layout.class.getName(), actionRequest);
 
+			if (layout.isTypeAssetDisplay() ||
```
Comment (ealonso) [inline modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/portlet/action/EditLayoutMVCActionCommand.java:129]: Hey @rolandpakai, an asset display page can also have draft layout, so, I think you should change the condition

Comment (rolandpakai) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/portlet/action/EditLayoutMVCActionCommand.java:129]: Hi @ealonso 

thanks for reviewing the fix. Could you please propose the change?
The fix was created with the help of @balazssk and we don't understand what you mean.
Could you please write a screnario where this case occures?

Thanks, Roland

Comment (ealonso) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/portlet/action/EditLayoutMVCActionCommand.java:129]: @rolandpakai 

I think it should be like this:

```
(layout.isTypeAssetDisplay() || layout.isTypeContent()) && (layout.fetchDraftLayout() == null)
```

Comment (rolandpakai) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/portlet/action/EditLayoutMVCActionCommand.java:129]: @ealonso 
We used the condition from here:
https://github.com/liferay/liferay-portal/blob/master/modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/frontend/taglib/form/navigator/LayoutDetailsFormNavigatorEntry.java#L49-L57

**Comment by ealonso** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/portlet/action/EditLayoutMVCActionCommand.java (hunk 1 above)
Comment (ealonso) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/portlet/action/EditLayoutMVCActionCommand.java:129]: Thanks for letting me know, this check is also wrong. I'm going to update it also

**Comment by ealonso** [comment]
Please see: https://github.com/liferay-echo/liferay-portal/pull/7761

**Pull request closed without merging**